### PR TITLE
validate duplicate files

### PR DESCRIPTION
**Pull request by leoraba**
MD5 file validation on submission
issue #824

### CODE REVIEW

File: song-server/src/main/java/bio/overture/song/server/controller/SubmitController.java [hunk 1]
```diff
@@ -55,7 +50,8 @@ public class SubmitController {
   public SubmitResponse submit(
       @RequestHeader(value = AUTHORIZATION, required = false) final String accessToken,
       @PathVariable("studyId") String studyId,
+      @RequestParam(value = "ignoreDuplicates", defaultValue = "false") boolean ignoreDuplicates,
```
Comment (leoraba) [inline song-server/src/main/java/bio/overture/song/server/controller/SubmitController.java:53]: ignoreDuplicates set to false by default

File: song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java [hunk 2]
```diff
@@ -348,6 +348,18 @@ public void checkAnalysisAndStudyRelated(@NonNull String studyId, @NonNull Strin
     }
   }
 
+  @Override
+  public void checkDuplicateAnalysis(@NonNull Payload payload) {
+    payload
+        .getFiles()
+        .forEach(
+            file -> {
+              if (fileService.checkMd5FileExists(file.getFileMd5sum())) {
```
Comment (leoraba) [inline song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java:357]: checking md5 of every file on Payload

File: song-server/src/main/java/bio/overture/song/server/controller/SubmitController.java (hunk 1 above)
Comment (justincorrigible) [reply song-server/src/main/java/bio/overture/song/server/controller/SubmitController.java:53]: for my own clarity, "ignore" means allow? or does it skip them?

Comment (leoraba) [reply song-server/src/main/java/bio/overture/song/server/controller/SubmitController.java:53]: **ignoreDuplicates** as true will "allow" to submit duplicate analysis.

**Review by justincorrigible — APPROVED** [review]
great work, as always!

**Pull request merged**